internal/shortener/repository/repository.go

```diff
@@ -280,92 +288,148 @@ func NewStoreDB(cfg config.Config) (*StoreDB, error) {
+func (store *StoreDB) GetShortener(code string) (model.Shortener, error) {
 	var url string
-	row := s.database.QueryRow(
-		"SELECT url FROM shortener"+
+	var delFlag bool
+	row := store.database.QueryRow(
+		"SELECT url, del_flag FROM shortener"+
 			" WHERE code = $1",
-		req.Code)
-	err := row.Scan(&url)
+		code)
+	err := row.Scan(&url, &delFlag)
 	if err != nil {
-		return nil, err
+		return model.Shortener{}, err
+	}
+	if delFlag {
+		return model.Shortener{}, ErrGetShortenerGone
 	}
 
-	return &GetShortenerResponse{
-		URL: url,
+	return model.Shortener{
+		Key:  model.ShortenerKey{Code: code},
+		Data: model.ShortenerData{URL: url},
 	}, nil
 }
 
-func (s *StoreDB) SetShortener(ctx context.Context, req *SetShortenerRequest) (*SetShortenerResponse, error) {
+func (store *StoreDB) SetShortener(ctx context.Context, s model.Shortener) (model.Shortener, error) {
 	// Проверка: уже существует
 	var oldCode string
-	row := s.database.QueryRowContext(ctx,
+	row := store.database.QueryRowContext(ctx,
 		"SELECT code FROM shortener"+
 			" WHERE url = $1",
-		req.URL)
+		s.Data.URL)
 	err := row.Scan(&oldCode)
 	if err == nil { // как ловить именно пустой результат, а не все ошибки БД? Ошибка нетипизирована error(*errors.errorString) *{s: "sql: no rows in result set"}
-		return &SetShortenerResponse{
-			Code: oldCode,
-			URL:  req.URL,
+		return model.Shortener{
+			Key:  model.ShortenerKey{Code: oldCode},
+			Data: model.ShortenerData{URL: s.Data.URL},
 		}, ErrSetShortenerAlreadyExists
 	}
 
-	query := "INSERT INTO shortener (code, url)" +
-		" VALUES ($1, $2)" +
+	query := "INSERT INTO shortener (code, url, uuid)" +
+		" VALUES ($1, $2, $3)" +
 		" ON CONFLICT (code) DO NOTHING"
-	_, err = s.database.ExecContext(ctx, query, // не понял как вернуть отсюда конфликтующую строку. Returning при конфликте возвращает пустоту
-		req.Code, req.URL)
+	_, err = store.database.ExecContext(ctx, query, // не понял как вернуть отсюда конфликтующую строку. Returning при конфликте возвращает пустоту
+		s.Key.Code, s.Data.URL, s.Data.User)
 	if err != nil {
-		return nil, err
+		return model.Shortener{}, err
 	}
 
-	return &SetShortenerResponse{
-		Code: req.Code,
-		URL:  req.URL,
-	}, nil
+	return s, nil
 }
 
-func (s *StoreDB) SetShortenerBatch(ctx context.Context, req *SetShortenerRequestBatch) (*SetShortenerResponseBatch, error) {
+func (store *StoreDB) SetShortenerBatch(ctx context.Context, s []model.Shortener) ([]model.Shortener, error) {
 
-	tx, err := s.database.Begin()
+	tx, err := store.database.Begin()
 	if err != nil {
 		return nil, err
 	}
 
-	var resp SetShortenerResponseBatch
-	for _, req := range req.Rows {
+	var respSBatch []model.Shortener
+	for _, reqS := range s {
 		// Проверка: уже существует
 		var oldCode string
-		row := s.database.QueryRowContext(ctx,
+		row := store.database.QueryRowContext(ctx,
 			"SELECT code FROM shortener"+
 				" WHERE url = $1",
```

Comment:
```suggestion
				" WHERE url = $1 FOR UPDATE",
```
https://t.me/bushigo/36